source/containers/Form/hooks/textReplacement.ts

```diff
@@ -190,136 +205,83 @@ export const replaceText = (
-        }
-
-        if (qs.items) {
-          qs.items = qs.items.map((item) => ({
-            ...item,
-            title: replaceText(
-              item.title,
-              user,
-              period,
-              partner,
-              encryptionPin
-            ),
-          }));
-        }
-
-        if (qs.categories) {
-          qs.categories = qs.categories.map((category) => ({
-            ...category,
-            description: replaceText(
-              category.description,
-              user,
-              period,
-              partner,
-              encryptionPin
-            ),
-          }));
-        }
-
-        if (qs.inputs) {
-          qs.inputs = qs.inputs.map((input) => ({
-            ...input,
-            label:
-              input.label &&
-              replaceText(input.label, user, period, partner, encryptionPin),
-            title:
-              input.title &&
-              replaceText(input.title, user, period, partner, encryptionPin),
-          }));
-        }
-
-        if (qs.components) {
-          qs.components = qs.components.map((input) => ({
-            ...input,
-            text: replaceText(input.text, user, period, partner, encryptionPin),
-          }));
-        }
-
-        return qs;
-      });
-    }
-    if (step.title)
-      step.title = replaceText(
-        step.title,
-        user,
-        period,
-        partner,
-        encryptionPin
-      );
-    if (step.description)
-      step.description = replaceText(
-        step.description,
-        user,
-        period,
-        partner,
-        encryptionPin
-      );
-    return step;
+  const replaceString = (text: string | undefined): string =>
+    text
+      ? replaceText(
+          text,
+          user,
+          period,
+          partner,
+          encryptionPin,
+          completionsClarificationMessage
+        )
+      : "";
+
+  return steps.map((step) => {
+    const stepTitle = replaceString(step.title);
+    const stepDescription = replaceString(step.description);
+
+    let questionTextReplaced: Question[] = [];
+
+    questionTextReplaced = (step.questions ?? []).map((question: Question) => ({
+      ...question,
+      text: replaceString(question.text),
+      label: replaceString(question.label),
+      title: replaceString(question.title),
+      heading: replaceString(question.heading),
+      items: (question.items ?? []).map((item) => ({
+        ...item,
+        title: replaceString(item.title),
+      })),
+      categories: (question.categories ?? []).map((category) => ({
+        ...category,
+        title: replaceString(category.description),
+      })),
+      inputs: (question.inputs ?? []).map((input) => ({
+        ...input,
+        label: replaceString(input.label),
+        title: replaceString(input.title),
+      })),
+    }));
+
+    return {
+      ...step,
+      title: stepTitle,
+      description: stepDescription,
+      questions: questionTextReplaced,
+    };
```

Comment:
This could perhaps be even more refactored like (notice the return property names):

```js
return steps.map((step) => {
  const title = replaceString(step.title);
  const description = replaceString(step.description);
  const questions = (step.questions ?? []).map(...);

  return {
    ... step,
    title,
    description,
    questions
  }
});
```